runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1827,6 +1945,24 @@ Java_java_lang_Class_getNestHostImpl(JNIEnv *env, jobject recv)
 
 	J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
 	J9Class *nestHost = clazz->nestHost;
+
+	/*
+	 * Nest host field is set post-verification, during an access check or during
+	 * a call to relevant java.lang.Class API method via the loadAndCheckNestHost
+	 * function. If loadAndCheckNestHost fails to return a valid nest host, then
+	 * getNestHostImpl returns this.
+	 */
+	if (NULL == nestHost) {
+		/* If loadAndCheckNestHost can not successfully load & verify a class's
+		 * nest host, then it will return the class - as required by the
+		 * getNestHost spec.
+		 */
+		nestHost = loadAndCheckNestHost(currentThread, clazz, FALSE);
+		if (NULL == nestHost) {
```

Comment:
if `loadAndCheckNestHost` is going to return `clazz` on unsuccessful load and verify, then we don't need this `if` block.